Lib9c/Action/RankingBattle.cs

```diff
@@ -146,17 +152,152 @@ public override IAccountStateDelta Execute(IActionContext context)
+                    simulator.Log.result,
+                    simulator.Log.Count
+                );
+                sw.Restart();
+
+                foreach (var itemBase in simulator.Reward.OrderBy(i => i.Id))
+                {
+                    Log.Verbose(
+                        "{AddressesHex}RankingBattle Add Reward Item({ItemBaseId}): {Elapsed}",
+                        addressesHex,
+                        itemBase.Id,
+                        sw.Elapsed);
+                    avatarState.inventory.AddItem(itemBase);
+                }
+
+                sw.Stop();
+                Log.Verbose("{AddressesHex}RankingBattle Serialize WeeklyArenaState: {Elapsed}", addressesHex, sw.Elapsed);
+                sw.Restart();
+
+                states = states
+                    .SetState(inventoryAddress, avatarState.inventory.Serialize())
+                    .SetState(arenaInfoAddress, arenaInfo.Serialize())
+                    .SetState(enemyInfoAddress, enemyInfo.Serialize())
+                    .SetState(questListAddress, avatarState.questList.Serialize());
+
+                if (migrationRequired)
+                {
+                    states = states
+                        .SetState(worldInformationAddress, avatarState.worldInformation.Serialize())
+                        .SetState(avatarAddress, avatarState.SerializeV2());
+                }
+
+                if (listCheck)
+                {
+                    var addressList = states.TryGetState(addressListAddress, out List rawAddressList)
+                        ? rawAddressList.ToList(StateExtensions.ToAddress)
+                        : new List<Address>();
+
+                    if (!addressList.Contains(avatarAddress))
+                    {
+                        addressList.Add(avatarAddress);
+                    }
+
+                    if (!addressList.Contains(enemyAddress))
+                    {
+                        addressList.Add(enemyAddress);
+                    }
+
+                    states = states.SetState(addressListAddress,
+                        addressList.Aggregate(List.Empty,
+                            (current, address) => current.Add(address.Serialize())));
+                }
+                sw.Stop();
+                Log.Verbose("{AddressesHex}RankingBattle Serialize AvatarState: {Elapsed}", addressesHex, sw.Elapsed);
+                sw.Restart();
+
+                var ended = DateTimeOffset.UtcNow;
+                Log.Verbose("{AddressesHex}RankingBattle Total Executed Time: {Elapsed}", addressesHex, ended - started);
+                EnemyPlayerDigest = enemyPlayerDigest;
+                return states;
+            }
+            // Run Backward compatible
+            return BackwardCompatibleExecute(rawWeeklyArenaState, sheets, avatarState, costumeStatSheet, sw, addressesHex, enemyAvatarState, ctx, states, inventoryAddress, questListAddress, migrationRequired, worldInformationAddress, started);
```

Comment:
How about inverting [if condition](https://github.com/planetarium/lib9c/pull/897/files#diff-12fde98e1392700396baaa420aa9fa7f535f5d36ddc14fa61d8305ef5486c1b4R160) and returning early?